crucible-mir/src/Mir/Trans.hs

```diff
@@ -1112,17 +1118,23 @@ evalPlaceProj ty pl@(MirPlace tpr ref NoMeta) M.Deref = do
         CTyBox t -> doRef t
         _ -> mirFail $ "deref not supported on " ++ show ty
   where
-    doRef (M.TySlice _) | MirSliceRepr tpr' <- tpr = doSlice tpr' ref
-    doRef M.TyStr | MirSliceRepr tpr' <- tpr = doSlice tpr' ref
+    doRef (M.TySlice ty') | MirSliceRepr <- tpr = doSlice ty' ref
+    doRef M.TyStr | MirSliceRepr <- tpr = doSlice (M.TyUint M.B8) ref
     doRef (M.TyDynamic _) | DynRefRepr <- tpr = do
         dynRef <- readMirRef tpr ref
         return $ MirPlaceDynRef dynRef
-    doRef _ | MirReferenceRepr tpr' <- tpr = do
+    doRef ty' | MirReferenceRepr <- tpr = do
+        -- This use of `tyToRepr` is okay because `TyDynamic` and other unsized
+        -- cases are handled above.
+        Some tpr' <- tyToReprM ty'
         MirPlace tpr' <$> readMirRef tpr ref <*> pure NoMeta
     doRef _ = mirFail $ "deref: bad repr for " ++ show ty ++ ": " ++ show tpr
 
-    doSlice tpr' ref' = do
-        slice <- readMirRef (MirSliceRepr tpr') ref'
+    doSlice ty' ref' = do
+        -- This use of `tyToRepr` is okay because we know the element type of a
+        -- slice is always sized.
+        Some tpr' <- tyToReprM ty'
```

Comment:
```suggestion
    doSlice ty' ref' = do
        -- This use of `tyToReprM` is okay because we know the element type of a
        -- slice is always sized.
        Some tpr' <- tyToReprM ty'
```